File: frontend/src/types/merged-model.ts
export interface MergedModel {
  readonly id: string;
  readonly projectId: string;
  readonly baseModelId: string;
  readonly sourceRunId: string | null;
  readonly adapterStep: number | null;
  readonly filePath: string;
  readonly fileSizeBytes: number;
  readonly createdAt: string;
}

export interface MergedModelList {
  readonly items: ReadonlyArray<MergedModel>;
  readonly total: number;
}

export interface MergeRunRequest {
  readonly sourceRunId: string;
}

File: backend/app/schemas/merged_model.py
from __future__ import annotations

from pydantic import BaseModel


class MergedModelResponse(BaseModel):
    id: str
    project_id: str
    base_model_id: str
    source_run_id: str | None
    adapter_step: int | None
    file_path: str
    file_size_bytes: int
    created_at: str

    model_config = {"from_attributes": True}


class MergedModelListResponse(BaseModel):
    items: list[MergedModelResponse]
    total: int


class MergeRunRequest(BaseModel):
    source_run_id: str
